### ISSUE TITLE
Doesn't work at all for me

### ISSUE THREAD

**Issue by benlwilliams** [Schepp/box-sizing-polyfill#2]
Not sure if I'm doing something wrong, or if a code problem. 
I'm running IE9 and when I put it in Browser Mode: IE7, nothing happens to correct the box sizing. 

My file structure is: /ebe_simplified/js/libs/boxsizing.htc
My css files have the following after any box-sizing declaration: *behavior: url(ebe_simplified/js/libs/boxsizing.htc);

**Comment by Schepp** [comment]
Hi! What if you try:

```
*behavior: url(/ebe_simplified/js/libs/boxsizing.htc);
```

?

**Comment by benlwilliams** [comment]
I actually tried the leading slash first, but when that didn't work
tried removing the slash. Are there certain server requirements that
need to be in place first for this to work?

On Feb 22, 2012, at 11:00 AM, Christian Schaefer
reply@reply.github.com
wrote:

**Comment by benlwilliams** [comment]
Maybe it has something to do with me using percent widths.  I though I read
that your script can handle that, but maybe I misunderstood?

**Comment by Schepp** [comment]
/edit: Percent width is exactly the use case of box-sizing: border-box. So no, that is not the reason.

Maybe it has to do with a wrong MIME type. Try setting it by adding the following line in an .htaccess file within either your site's root or within _/ebe_simplified/js/libs/_:

```
AddType text/x-component .htc
```

Or, for whatever reason, IE doesn't fetch/find the .htc-file which you could check in a network graph like dynaTrace AJAX Edition.

**Comment by benlwilliams** [comment]
Thanks for the suggestion, but unfortunately that didn't work either.  Any
other ideas?

On Wed, Feb 22, 2012 at 3:14 PM, Christian Schaefer <
reply@reply.github.com

**Comment by Schepp** [comment]
Did you check what happens inside [dynaTrace AJAX Edition](http://ajax.dynatrace.com/ajax/en/)?

**Comment by benlwilliams** [comment]
Never heard of it, but I'll install it and see if it reveals anything.
Thanks.

On Wed, Feb 22, 2012 at 3:26 PM, Christian Schaefer <
reply@reply.github.com

**Comment by adamnbowen** [comment]
@Schepp I'm also having trouble getting this to work, do you happen to have an example of this working in the wild?  [Here is the site I'm trying to get it working on](http://erisa.adsincchicago.com) (only has minimal testing, needs refactoring, etc.).

I can see the /css/behavior/boxsizing.htc file show up on the Network tab for ie7 with ie9's developer tools, so I know it is loading the file at least (and it does come down as text/x-component type), but the layout remains broken.

Any ideas? Don't know if this is the same root cause of @blw0010, since he didn't link an actual site.

**Comment by benlwilliams** [comment]
I too can see the htc file, but the actual box sizing fix never takes place.

On Feb 28, 2012, at 8:19 AM, Adam Bowen
reply@reply.github.com
wrote:

**Comment by adamnbowen** [comment]
@blw0010 Does removing the checks for `if(typeof(element.style.boxSizing) == "undefined")` fix it for you? It resolved the issue for me.  Seems that ie7 (at least ie7 mode in ie9) reports boxSizing as something other than undefined.

Since we have the `*behavior` check in, we shouldn't have any ill effects from removing the checks from the script

**Comment by Schepp** [comment]
I just checked the site you mentioned, Adam, in IETester, and I witnessed my polyfill in action. So maybe you guys are having trouble with your IE9? Are you maybe confusing browser mode with document mode? Anyway, IE9 will never be able to mimic to 100% a real IE7, that's why if you want to make sure, you need to check with MS's virtual machines.

**Comment by Schepp** [comment]
Thanks Adam! Now I hope that everyboy follows my advice and uses `*behavior` so it doesn't break in IE8.

**Comment by adamnbowen** [comment]
@Schepp we _may_ be expecting too much of people ;).  I wonder if Modernizr can detect the lack of support properly... I'll do a bit of investigating.

**Comment by adamnbowen** [comment]
Okay, I've been playing with detection for `document.documentMode` to check for ie7 mode in ie9, and for whatever reason, I just can't wrap my head around the issue, so since I have a working solution already in place, I'm just going with that ;)

**Comment by benlwilliams** [comment]
Sorry I disappeared for a while. I got swallowed up by a freelance job and just resurfaced. The last merge that was made frmo adamnbowen made it work, sort of, but not really.  The vertical padding was still out of wack and things would grow/shrink when I moved the mouse in and out of the element.   This was happening in IE9 in IE7 browser mode, IE7 doctype. I just installed IEtester and it does not work in there either. Not at all. I can't seem to figure out what I'm doing wrong.

**Comment by Schepp** [comment]
Can you send me a link to the page in question?

**Comment by benlwilliams** [comment]
Sure.  Here it is:
http://bwilliams.dev.genares.net/ebe_simplified/template/room_list.php

On Wed, Mar 7, 2012 at 5:09 PM, Christian Schaefer <
reply@reply.github.com

**Comment by drublic** [comment]
@blw0010 Not sure where to find the right thing on that page but `.w-half` misses a `%` for it's `width` which kinda breaks the page for me on Chrome.
Furthermore I cannot find the linked `.htc`-file in your CSS. Am I missing something?

**Comment by benlwilliams** [comment]
Sorry. I had to give up in the htc file and move forward due to time
constraints. I'll try reverting and fixing the % to see how it affects
things.

Sent from my iPhone
469-387-4998

On Mar 11, 2012, at 11:14 AM, Hans
reply@reply.github.com
wrote:

**Comment by benlwilliams** [comment]
When I did have the .htc file linked correctly, and was getting partial results, I noticed that my jQuery animations were very choppy and slow to redraw the layout.  Is this a known issue with this polyfill?

**Comment by exside** [comment]
lol...I'm working on a project where I used the old box-sizing.htc polyfill from 2002 (or so) without the % capability...so it wasn't working for me and I searched hours for a solution...but I didn't find this updated version here _hairpulling_...didn't try it yet because I then found an ugly fix that made it work in IE6/7 and this is just the (bad) old quirks mode triggered by a comment before the HTML5 Doctype... it looks fine in these dinosaur browsers and surprisingly quirks mode isn't triggered in IE8+ or any other browser I checked (it should in my opinion but it isn't...can maybe somebody enlighten me why this isn't the case...it's very good that it doesn't happen, but I don't understand it^^). Now I have to check if this one works...thanks for it anyways!

**Comment by exside** [comment]
It's working for me now in every browser down to IE6...I took a step back and tried again with Schepps box-sizing.htc and it works!

**Comment by Schepp** [comment]
BTW, @lluuuk that comment thing is an awesome finding/trick. Thanks for sharing!

**Comment by exside** [comment]
I have to thank you for all your work I regularly use^^...so thanks for Booster (which I turned into an amateur Modx Snippet, but it works on most of my sites =D), css prefex spawner and the box-sizing.htc!

keep it up!

kind regards from switzerland!

PS: Any ideas why the comment thing does work the way it does? Shouldn't it trigger quirksmode in all browsers??

**Comment by jkd77** [comment]
I also cannot seem to get this to work. If anyone has any thoughts on what I'm doing wrong I'd love to hear them. I have a simple test case here: http://testing.werremeyer.com/html/templates08d/xhtml/widthTest.html

I've been using Adobe Browserlab to test IE7 since I work on an older Mac. Not sure if the htc fixes would show up there, but I can set a number of seconds to wait until the screenshot is taken.

I also added the .htaccess file as suggested. Thoughts?

**Comment by exside** [comment]
It does work =)...so something in your setup seems to be wrong...maybe the absolute URL to the .htc file? Don't know if IE likes that...the relative URL has to be relative to the .html file and not to the .css file...in case you want to test it with a relative path

**Comment by Schepp** [comment]
And please test in a real IE7, not something that pretends to be IE7. Use those images here:
https://github.com/xdissent/ievms

**Comment by jkd77** [comment]
Figured it out!!! The % sized columns effected by border-box and therefore the polyfill fix are not sized correctly in IE7 whenever the columns' parent container is sized using a pixel max-width - as in max-width: 940px. Using a set pixel width works - width: 940px. It would be really cool if it could work in either scenario...  Thoughts?
